### PR TITLE
Ruby Docker runner

### PR DESCRIPTION
**Pull request by danielpclark**
I've abstracted away any need for extra Docker image layers.

Fully TDD.

OmniRunner is a superclass for creating individual Docker runners for various languages.  It should work with most languages with simply setting a few values and methods in a subclass.

### CODE REVIEW

**Comment by danielpclark** [comment]
Need to add Docker to TravisCI and create two test paths; one with Docker and one without.

**Comment by bokmann** [comment]
First pass grok review I like it.  I especially like how the RubyRunner clearly documents what is specific about ruby (the command line and the base image). I had some questions about how this design with OmniRunner would handle compile vs. build, especially in notifying failures.  Those were mostly answered.  When I asked 'how does this dovetail with what @yarmiganosca is working on?', there was conversation that indicated uncertainty and perhaps some overlap in what the two of you are doing.  Please coordinate and reconcile...  once the two of you are on the same page, I'm happy with what the two of you do based on this work and what I know of his.

**Comment by danielpclark** [comment]
Looking at @yarmiganosca's work they look like they'll be a good fit.  OmniRunner is actually more of a builder.  It doesn't execute the code itself as much as it produces the Docker command to execute.  What Chris has done looks like I can also do with this and wrap that abstraction around my builder/runner.  The OmniRunner follows the Single Responsibility Principle.  Perhaps I should rename it from Runner to Builder @bokmann ?

Once both this and @yarmiganosca code is merged in I can add the additional object API that Chris has implemented around my RubyRunner (RubyBuilder?).

**Comment by bokmann** [comment]
I'm ok iwith merging this... I like the design; I just want @yarmiganosca to weigh in in a comment as he's been closer to the docker grading work than I've been and this deserves a close pair of eyes.  I like the rename you suggest as well.

**Comment by danielpclark** [comment]
@bokmann this is ready.

File: spec/docker/ruby_builder_spec.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+          expect(File.exist? bad_entry.path).to be_truthy
+
+          builder = Docker::RubyBuilder.new(dir)
+
+          sys_exec(builder.build(good_entry, input, output))
+          expect(out).to eq('')
+          expect(err).to eq('')
+          expect(@exitstatus).to eq(0)
+
+          sys_exec(builder.build(bad_entry, input, output))
```
Comment (yarmiganosca) [inline spec/docker/ruby_builder_spec.rb]: if good fails, bad won't be tested. These should be separate examples

**Pull request merged**